src/main/java/com/commercetools/sync/products/utils/VariantReferenceReplacementUtils.java

```diff
@@ -128,11 +129,8 @@
     }
 
     private static Optional<Reference<Product>> getProductReference(@Nonnull final Attribute attribute) {
-        try {
-            return Optional.of(attribute.getValue(AttributeAccess.ofProductReference()));
-        } catch (final JsonException exception) {
-            return Optional.empty();
-        }
+        return isProductReference(attribute)
+            ? Optional.of(attribute.getValue(AttributeAccess.ofProductReference())) : Optional.empty();
```

Comment:
more functional style of the implementation:
```java
Optional.of(attribute)
    .filter(VariantReferenceReplacementUtils::isProductReference)
    .map(a -> a.getValue(AttributeAccess.ofProductReference()));
```

same for [`getProductReferenceSet()`](https://github.com/commercetools/commercetools-sync-java/pull/180/files#diff-c39e1d4e74d08c6b71a90fec82c72826R148) below

Reply:
5be11f0